include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_scan.h

```diff
@@ -127,13 +129,24 @@ single_pass_scan_impl(sycl::queue __queue, _InRange&& __in_rng, _OutRange&& __ou
     ::std::size_t num_workitems = num_wgs * wgsize;
 
     constexpr int status_flag_padding = SUBGROUP_SIZE;
-    std::uint32_t status_flags_size = num_wgs + status_flag_padding + 1;
-    std::uint32_t tile_sums_size = num_wgs + status_flag_padding;
+    std::size_t status_flags_elems = num_wgs + status_flag_padding + 1;
+    std::size_t status_flags_size = status_flags_elems * sizeof(std::uint32_t);
+
+    std::size_t tile_sums_elems = num_wgs + status_flag_padding;
+    std::size_t tile_sums_size = status_flags_elems * sizeof(std::uint32_t);
```

Comment:
```suggestion
    std::size_t tile_sums_size = status_flags_elems * sizeof(_Type);
```
Should this not be of the type size?